sunpy/instr/aia/response.py

```diff
@@ -0,0 +1,462 @@
+
+"""
+
+import numpy as np
+import pandas as pd
+from astropy import units as u
+from scipy.io import readsav
+
+import sunpy
+import sunpy.data.test as test
+
+
+class Response():
+    """
+    class description here!
+
+    Parameters
+    ----------
+    path_to_genx_file: string
+        give a path to a instrument file from .genx output to initialize a dataframe for the class
+    save_to_outfile: bool
+        loads channel properties into a dataframe and saves to a .csv file
+    csv: string
+        alternately, give a path to the csv file to read in data information where the keys are channel wavelength centers and the values are properties
+
+    Attributes
+    ----------
+    properties: array of strings
+        The names of properties that are imported from SSW
+    wavelength_centers: array of integers
+        The center wavelength of each wavelength range or channel imaged with AIA
+    dataframe: pandas dataframe
+        The property information stored after being imported from SSW
+    wavelength_range: array
+        the range of wavelengths imaged per channel with AIA
+
+
+    # Example:
+    # >>> from sunpy.instr.aia import Response
+    #
+    # # load in dataframe to be used
+    # >>> r = Response(path_to_genx_file='/media/gsoc/ssw_aia_response_data/aia_V6_all_fullinst', save_outfile=True)
+    #
+    # # call functions
+    # >>> r.get_properties_per_channel(94)
+
+
+    Notes:
+    Currently the instr directory contains the file aia.py (previously aiaprep.py - I think?)
+    This branch creates instr/aia directory and will have both this program response.py and aiaprep.py
+
+    Feedback/ Thoughts are always welcome! -Thanks!
+    """
+
+    def __init__(self, **kwargs):
+        # no keywords for now. emissivity, temperature, density possible later
+
+        # parameter list
+        self.properties = ['wave', 'effarea', 'units', 'geoarea', 'scale', 'platescale',
+                           'numfilters', 'wavemin', 'wavestep', 'wavenumsteps', 'wavelog',
+                           'filtersincludemesh', 'meshtrans', 'usecontam', 'contamthick',
+                           'usephottoelec', 'elecperev', 'usephottodn', 'elecperdn', 'useerror',
+                           'fp_filter', 'ent_filter', 'primary', 'secondary', 'ccd', 'contam',
+                           'cross_area']
+
+        # wavelength center list
+        # TODO: want 6 EUV channels and 2 UV channels --- needs updated
+        self.wavelength_centers = [94, 131, 171, 193, 211, 304, 335]
+
+        #           Notes: I needed a way to define self.dataframe locally inside the aia_inst_genx_to_dataframe
+        #           and afterwards use it globally. This is what i have found, but am open to other suggestions.
+
+        #           I could import os and recursivly search the .genx directory for the files needed rather than the user defining one file. Then, the variable would change to path_to_genx_dir
+
+
+        # keyword arguements
```

Comment:
You can do all of this a bit more compactly in the constructor. For example,
```Python
def __init__(self,path_to_genx_file='', save_outfile=False, csv=''):
```
The `kwargs` method and this way seem equivalent to me with the former being a bit more verbose. I'm not sure if there is a preferred style, either by Astropy/SunPy or the PSF. Thoughts on this @Cadair @drewleonard42 ?